libqtile/widget/widgetbox.py

```diff
@@ -138,7 +134,7 @@ def _configure(self, qtile, bar):
 
         # # Overwrite the current drawer.draw method with a no-op
         for w in self.widgets:
-            w.widget.drawer.draw = _no_draw
+            w.widget.drawer.draw = w.widget.drawer._check_surface_reset
```

Comment:
Hmm this has turned out a bit uglier than i expected. Maybe we should move this internal-to-Drawer logic within Drawer? i.e. here we could have:
```py
w.widget.draw.disable()  # or pause?
```
Then re-enable/unpause it when it gets shown. It would be a bit neater and more clear i reckon.

Then `Drawer` would, inside these two methods, override `self.draw` how it wants.

Reply:
Is there another use case for that? 

I agree it would be neater and maybe that's a good enough reason on its own!

Reply:
Not really, I was just triggered by the accessing of a private method of another class. I'm also thinking about long-term readability - the name of `_check_surface_reset` describes what it does but it isn't clear of why it's needed unless you go and find where it is used here.

Reply:
Well, I've changed it for you anyway!